google/api_versions.go

```diff
@@ -84,8 +86,34 @@ func (s Feature) HasChangeBy(d TerraformResourceData) bool {
 
 // Return true when a feature appears in schema or has been modified.
 func (s Feature) InUseBy(d TerraformResourceData) bool {
-	_, ok := d.GetOk(s.Item)
-	return ok || s.HasChangeBy(d)
+	return inUseBy(d, s.Item)
+}
+
+func inUseBy(d TerraformResourceData, path string) bool {
+	pos := strings.Index(path, "*")
```

Comment:
I have to say I'm not a huge fan of this syntax. I appreciate that you do have to make some sort of change however to be able to see if this feature is in use for any of the `network_interface` properties.

If I see a '*' I would expect it to be globbing like a regex rather than having it match a sublist. Anyway to make the behavior like that?

(Alternatively, if we could see if a feature is in use by using a more structural syntax I'd totally love that (similar to how we navigate the `schema.ResourceData` struct)

Also, can we document this syntax in the `inUseBy` method?

Reply:
There is no use case for globbing on a field. I don't expect why a beta field would be present under two different field. `inUseBy` is only use for api versions.

You can see the `*` as globbing on the indices. Usually, you would have `field.5.nested_field`.

I am not sure I follow your suggestion about structural syntax...

I will add more documentation once we agree on the syntax.

Reply:
But you're globbing on a field right now! (or at least, you're globbing on an integer)

I guess I'm saying is that if we introduce syntax like '*' but it is specific to integers, it's asking for a bug in the future when someone uses '*' like globbing. Given that globbing would work here, would it be too hard to add?

For a more structural syntax, I meant being able to receive and parse the actual `schema.ResourceData` struct using `GetOk` and other syntax to see if a given feature is in use, rather than using the serialize string data.

Reply:
Supporting globbing also on field names would definitely add complexity (and potentially bugs) when there is clearly no needs for this. However, I can make sure it fails if with a nice error message if someone uses the `*` on a field name instead of an indice. This way, we prevent somebody from misusing it without knowing they are misusing it.

Reply:
I don't want to block this whole PR on the syntax (especially since what's currently-implemented would work if * was a globbing symbol) but I'm starting to worry that this 'inUseBy' method is an anti-pattern. Right now it depends on the serialized state and not the provided `schema.ResourceData` which provides a well-defined set of methods for dealing with the schema. My understanding is that `isInUse` is used because it's convenient, but if we ever depend on values like optionals or validation then we'll regret using it here.

That said, we can probably have that discussion outside of this PR. If we document the '*' syntax I'm ok as-is.

Reply:
I'm a bit confused - `inUseBy` uses the methods from `schema.ResourceData`; `TerraformResourceData` is just an interface for the two methods `inUseBy` needs, since we don't need the rest of the behaviour of the class and are able to unit test this code w/ a mock of that interface.

Reply:
OK, after looking at the code, you were right and I was confused; it IS operating on `schema.ResourceData`; I had no idea you could directly access attributes like `foobar.#` to get the size of a list for example using the Get or GetOk method. For some reason I assumed it was using something similar to the the attributes map that we constantly assert on in unit tests.

Reply:
Documentation added.